unpacker.go

```diff
@@ -113,9 +119,16 @@ EachLayer:
 			return errors.Wrapf(err, "failed to stat snapshot %s", chainID)
 		}
 
-		labelOpt := snapshots.WithLabels(map[string]string{
-			"containerd.io/snapshot.ref": chainID,
-		})
+		// filters the provided annotations by removing any key which isn't a snapshot
+		// label. Snapshot labels have a prefix of "containerd.io/snapshot/".
+		labels := make(map[string]string)
+		for k, v := range desc.Annotations {
+			if strings.HasPrefix(k, inheritedLabelsPrefix) {
+				labels[k] = v
+			}
+		}
+		labels[labelSnapshotRef] = chainID
```

Comment:
We already have similar logic in [metadata.filterInheritedLabels](https://github.com/containerd/containerd/blob/master/metadata/snapshot.go#L946).
Would it make sense to move it to `snapshots` package and reuse it from there in both places?

Reply:
Thank you for your review. I fixed the filter implementation.